ethcontract-common/src/artifact/hardhat.rs

```diff
@@ -72,26 +77,44 @@ pub struct HardHatLoader {
     /// Deny list takes precedence over allow list. That is, if network
     /// appears in both, it will be denied.
     pub networks_deny_list: Vec<NetworkEntry>,
+
+    /// List of allowed contract names.
+    ///
+    /// When loading artifact, loader will only load contracts if their names
+    /// are present in this list.
+    ///
+    /// Empty list means that all contracts are allowed.
+    pub contracts_allow_list: Vec<String>,
+
+    /// List of denied contract names.
+    ///
+    /// When loading artifact, loader will not load contracts if their names
+    /// are present in this list.
+    ///
+    /// Empty list means that no contracts are denied.
+    pub contracts_deny_list: Vec<String>,
```

Comment:
Could you provide a reason or example use case for this?

Reply:
The issue, as I understand it, is that you can have contracts with the same name deployed with different bytecode across different networks with hardhat deployments (its not all or nothing like Truffle - which only keeps bytecode/ABI for the latest deployment). Those contracts need to be excluded because generating code for them is not possible (what if you have two functions with the same name but conflicting types in different ABIs for different versions of the same contract?).

Reply:
Exactly. I'll add this to documentation.